File: MPP_MusicFestival_Java/src/main/java/service/LogService.java
package service;

import exceptions.LogException;
import repository.OfficeEmployeeRepository;

import java.util.Objects;

public class LogService {

    private final OfficeEmployeeRepository officeEmployeeRepository;

    public LogService(OfficeEmployeeRepository officeEmployeeRepository) {
        this.officeEmployeeRepository = officeEmployeeRepository;
    }

    private Long employeeId = null;

    public Long getEmployeeId() {
        return employeeId;
    }

    public Long logIn(String username, String password) throws LogException {
        if (employeeId != null) {
            throw new LogException("Already logged in!");
        }
        Long id = officeEmployeeRepository.getId(username);

        if (id==null) {
            throw new LogException("Wrong username or password!");
        }

        if (Objects.equals(officeEmployeeRepository.getPassword(id), password)) {
            employeeId = id;
        } else {
            throw new LogException("Wrong username or password!");
        }

        return id;
    }

    public void logOut() throws LogException {
        if (employeeId == null) {
            throw new LogException("You are not logged in!");
        }
        employeeId = null;
    }
}

File: MusicFestival/Service/LogService.cs
using MPP_MusicFestival.Repository;
using MusicFestival.Exceptions;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MusicFestival.Service
{
    public class LogService
    {
        private OfficeEmployeeRepository officeEmployeeRepository;

        public LogService(OfficeEmployeeRepository officeEmployeeRepository)
        {
            this.officeEmployeeRepository = officeEmployeeRepository;
        }

        public long employeeId = 0;

        public long getEmployeeId()
        {
            return employeeId;
        }

        public long logIn(string username, string password)
        {

        if (employeeId != 0)
                throw new LogException("Already logged in!");
    
        long id = officeEmployeeRepository.getId(username);

        if (id==0)
                throw new LogException("Wrong username or password!");

        if (officeEmployeeRepository.getPassword(id) == password)
                employeeId = id;
        else
                throw new LogException("Wrong username or password!");
        

        return id;
    }

    public void logOut()
    {
        if (employeeId == 0)
                throw new LogException("You are not logged in!");
        
        employeeId = 0;
    }

    }
}
